core/src/main/java/org/elasticsearch/action/support/nodes/TransportNodesAction.java

```diff
@@ -144,7 +148,7 @@ public void run() {
                         // those (and they randomize the client node usage, so tricky to find when)
                         onFailure(idx, nodeId, new NodeShouldNotConnectException(clusterService.localNode(), node));
                     } else {
-                        NodeRequest nodeRequest = newNodeRequest(nodeId, request);
```

Comment:
If you want to keep the immutability you can make ChildTask another class that wraps a regular task. Then you can wrap a convenient layer and you can keep the safe, comforting immutability. It costs you another object but I think that is worth it?